### ISSUE TITLE
Device Code Flow

### ISSUE THREAD

**Issue by lewis-green** [goauthentik/authentik#4040]
**Describe your question/**
An example of how to use the device flow. I have setup a flow and set it to be used on the tenant, but I cannot get a provider to work with device flow. 

Is there some steps on how to setup or what am I missing?

**Relevant infos**
2022.10.0

**Version and Deployment (please complete the following information):**
 - authentik version: [e.g. 2022.10.0]
 - Deployment: [e.g. helm]

**Additional context**

**Comment by Nayruden** [comment]
It took me embarrassingly long to figure out how to get this to work, so I'm leaving these instructions to save folks some time in the future.

First, activate the device code flow for the tenant by going to System -> Tenants. Edit your default tenant, expand the default flows, and set "Device code flow". I used "Pre-Authentication (default-source-pre-authentication)".

Then you can configure a provider and application like usual. The provider needs to be OAuth2. I set my "Redirect URI" for the provider to `*` but I'm not sure if that's necessary.

That's it! Now you can point any OAuth2 client that supports device flow at Authentik. The endpoint is `https://my_authentik.my_tld/application/o/device/`. For example, to test out the first step via curl...

```bash
curl -sd "client_id=MY-PROVIDER-CLIENT-ID&scope=openid email" -H "Content-Type: application/x-www-form-urlencoded" https://AUTHENTIK/application/o/device/
```

**Comment by heoyongun** [comment]
> It took me embarrassingly long to figure out how to get this to work, so I'm leaving these instructions to save folks some time in the future.
> 
> First, activate the device code flow for the tenant by going to System -> Tenants. Edit your default tenant, expand the default flows, and set "Device code flow". I used "Pre-Authentication (default-source-pre-authentication)".
> 
> Then you can configure a provider and application like usual. The provider needs to be OAuth2. I set my "Redirect URI" for the provider to `*` but I'm not sure if that's necessary.
> 
> That's it! Now you can point any OAuth2 client that supports device flow at Authentik. The endpoint is `https://my_authentik.my_tld/application/o/device/`. For example, to test out the first step via curl...
> 
> ```shell
> curl -sd "client_id=MY-PROVIDER-CLIENT-ID&scope=openid email" -H "Content-Type: application/x-www-form-urlencoded" https://AUTHENTIK/application/o/device/
> ```

I am struggling with the same issue.
#5566 

The following is working
http://localhost:9000/application/o/device/

But when I do the following to get the token, I get an error.
http://localhost:9000/application/o/token/

Did you solve the problem?


http://localhost:9000/application/o/device/

**Comment by TheDevMinerTV** [comment]
> It took me embarrassingly long to figure out how to get this to work, so I'm leaving these instructions to save folks some time in the future.
> 
> First, activate the device code flow for the tenant by going to System -> Tenants. Edit your default tenant, expand the default flows, and set "Device code flow". I used "Pre-Authentication (default-source-pre-authentication)".
> 
> Then you can configure a provider and application like usual. The provider needs to be OAuth2. I set my "Redirect URI" for the provider to `*` but I'm not sure if that's necessary.
> 
> That's it! Now you can point any OAuth2 client that supports device flow at Authentik. The endpoint is `https://my_authentik.my_tld/application/o/device/`. For example, to test out the first step via curl...
> 
> ```shell
> curl -sd "client_id=MY-PROVIDER-CLIENT-ID&scope=openid email" -H "Content-Type: application/x-www-form-urlencoded" https://AUTHENTIK/application/o/device/
> ```

Thank you, mate. Took me embarrassingly long to figure out as well. I was about to open an issue. This needs to get documented properly though.

**Comment by BeryJu** [comment]
@TheDevMinerTV this was caused by a bug that was fixed by #9076
